backend/app.py

```diff
@@ -2071,6 +2078,107 @@ def process_financial_documents():
             "status": "error",
             "message": str(e)
         }), 500
-    
+
+# main.py
+from app_config import IMAGE_PATH
+from summarization import DocumentSummarizer
+
+@app.route('/api/SECEdgar/summarize', methods=['GET'])
+def generate_summary():
+    # payload example:
+    # {
+    #     "equity_name": "MS",
+    #     "financial_type": "10-K"
+    # }
+    try:
+        data = request.get_json()
+        equity_name = data.get('equity_name')
+        financial_type = data.get('financial_type')
+
+        # Enhanced input validation
+        if not isinstance(equity_name, str) or not isinstance(financial_type, str):
+            return jsonify({
+                'error': 'Invalid input type',
+                'details': 'equity_name and financial_type must be strings'
+            }), 400
+
+        if not equity_name.strip() or not financial_type.strip():
+            return jsonify({
+                'error': 'Empty input',
+                'details': 'equity_name and financial_type cannot be empty'
+            }), 400
+
+        # Initialize components with error handling
+        try:
+            blob_manager = BlobStorageManager()
+            summarizer = DocumentSummarizer()
+        except Exception as e:
```

Comment:
@namtran6701  All unexpected errors return 500 Internal Server Error. Consider distinguishing between different error types (e.g., a custom exception when documents cannot be found or when the summarizer fails).